src/main/kotlin/com/angels/challenge/controller/TestController.kt

```diff
@@ -0,0 +1,68 @@
+package com.angels.challenge.controller
+
+import com.angels.challenge.repository.*
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.hateoas.MediaTypes
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.*
+
+
+@RestController
+class TestController() {
+    @Autowired
+    var repository: ClusterRepository? = null
+    @Autowired
+    var repositoryRegions: RegionRepository? = null
+    @Autowired
+    var storeRepository: StoreRepository? = null
+    @Autowired
+    var productRepository: ProductRepository? = null
+    @Autowired
+    var seasonRepository: SeasonRepository? = null
+    @Autowired
+    var storeSeasonRepository: StoreSeasonRepository? = null
+
+    @GetMapping("/clusters", produces = [MediaTypes.HAL_JSON_VALUE, MediaType.APPLICATION_JSON_VALUE])
+    @ResponseBody
+    fun getClusters(): ResponseEntity<Any?> {
+        val list = repository?.findAll()
+        return ResponseEntity(list, HttpStatus.OK)
+    }
+
+    @GetMapping("/regions", produces = [MediaTypes.HAL_JSON_VALUE, MediaType.APPLICATION_JSON_VALUE])
+    @ResponseBody
+    fun getRegions(): ResponseEntity<Any?> {
+        val list = repositoryRegions?.findAll()
+        return ResponseEntity(list, HttpStatus.OK)
+    }
+
+    @GetMapping("/stores", produces = [MediaTypes.HAL_JSON_VALUE, MediaType.APPLICATION_JSON_VALUE])
+    @ResponseBody
+    fun getStores(): ResponseEntity<Any?> {
```

Comment:
Could this signature be improved?

Reply:
the type of ResponseEntity could be StoreEntity or Store (assuming we would use a service and not communicate directly with the persistance layer)

Reply:
Actually, unless you have some configuration different than usual, you should be fine with something like just

`fun getStores(): Collection<Store> { `

or if returns just a single store:

`fun getStores(): Store { `

This apparently innocuous change can be critical if you embed into your Spring project an [OpenAPI generator](https://www.baeldung.com/spring-rest-openapi-documentation) that generates API documentation from your method signatures.